File: chap7/play-jbook/app/controllers/Application.java
package controllers;

import play.*;
import play.mvc.*;

import views.html.*;

import java.util.*;
import static java.util.Arrays.asList;

import static scala.collection.JavaConversions.collectionAsScalaIterable;

import sample.SampleUsers;

import models.*;

import org.joda.time.LocalDate;
import org.joda.time.LocalTime;
import org.joda.time.Days;

public class Application extends Controller {

  /**
   *  JavaScript version of the router to be injected in the main page
   */
  public static Result js() {
    response().setContentType("text/javascript");
    return ok(
      Routes.javascriptRouter("playRouter",
        // Routes
        controllers.routes.javascript.Chats.receiveImage(),
        controllers.routes.javascript.Chats.chatsStream(),
        controllers.routes.javascript.Real.so_web(),

        controllers.routes.javascript.Twitter.mentioning(),
        controllers.routes.javascript.Twitter.searchTag()
      )
    );
  }

  public static Result index() {
    String email = session().get("email");
    if (email != null) {
      return redirect( routes.Dashboard.index() );
    } else {
      return unauthorized(
        views.html.login.render()
      );
    }
  }

  public static Result login() {
    return ok(
        views.html.login.render()
    );
  }

  public static Result logout() {
    session().clear();
    return ok(
        views.html.login.render()
    );
  }

  public static Result enter() {
    Map<String,String[]> params;
    params = request().body().asFormUrlEncoded();

    String email = params.get("email")[0];
    User user = User.find.byId(email);
    if (user == null) {
      return redirect( routes.Application.login() );
    } else {
      session("email", email);
      return redirect( routes.Dashboard.index() );
    }
  }


}

File: chap8/play-sbook/app/controllers/Application.scala
package controllers

import play.api._
import play.api.mvc._

import models.Item
import org.joda.time.DateTime
import org.joda.time.Days
import org.joda.time.LocalTime

import models.User
import models.Item
import models.Chat

object Application extends Controller {

  /**
   *  JavaScript version of the router to be injected in the main page
   */
  def js = Action {implicit request =>
    Ok(
      Routes.javascriptRouter("playRouter")(
        // Routes
        routes.javascript.Chats.receiveImage,
        controllers.routes.javascript.Chats.chatsStream,

        controllers.routes.javascript.Twitter.mentioning,
        controllers.routes.javascript.Twitter.searchTag
      )
    ).as("text/javascript")
  }


  def index = Action { implicit request =>
    val email = request.session.get("email");
    email.map { _ =>
      Redirect( routes.Dashboard.index() )
    } getOrElse {
      Unauthorized(views.html.login())
    }
  }

  def login = Action { implicit request =>
    Ok(views.html.login())
  }

  def logout = Action { implicit request =>
    Ok(
        views.html.login()
    ).withNewSession
  }

  def enter = Action { implicit request =>
    (for {
      f <- request.body.asFormUrlEncoded;
      emails <- f.get("email");
      email <- emails.headOption;
      user <- User.load(email)
    } yield
      Redirect( routes.Dashboard.index() ).withSession(
        session + ("email" -> email)
      )
    ).getOrElse {
      Redirect(routes.Application.login())
    }
  }

}